components/brave_wallet/resources/brave_wallet_provider.js

```diff
@@ -60,4 +60,27 @@
     configurable: true,
     writable: true,
   })
+}
+
+function solanaSetup() {
+  if (!window.solana) {
```

Comment:
when is `window.solana` defined?

Reply:
same as `window.ethereum`
https://github.com/brave/brave-core/pull/12554/commits/9a1c8632d83d3a95e41de733a00c8fb0ca6cf421#diff-bd0508db5780666e635128717e778f2a51486f055feb9f236d03aefdb1dc122cR54-R57

Reply:
just to confirm, does this only get called when wallet is enabled? i.e., not in private windows or if wallet is disabled

Reply:
yes, in private window or wallet is disabled, both `window.ethereum` and `window.solana` are not available